src/Text/Pandoc/Readers/Org.hs

```diff
@@ -472,7 +483,7 @@ rawBlockContent (indent, blockType) = try $
   unlines . map commaEscaped <$> manyTill indentedLine blockEnder
  where
    indentedLine = try $ ("" <$ blankline) <|> (indentWith indent *> anyLine)
-   blockEnder = try $ indentWith indent *> stringAnyCase ("#+end_" <> blockType)
+   blockEnder = try $ skipSpaces *> stringAnyCase ("#+end_" <> blockType)
```

Comment:
I find it very, very tempting to deviate from standard Emacs here, I
feel like nested blocks would be much nicer.  However, you are
absolutely right in that personal sense of æsthetics shouldn't get in
the way of proper functionality.

I checked again, the Org docs don't say anything about nested code
blocks.  The Emacs-way of expressing this would be to use the following
'*' escape syntax.  Which, I just realized, isn't supported by the
reader yet.

    #+BEGIN_SRC org
        *#+BEGIN_SRC haskell
        putStrLn "Hello"
        *#+END_SRC
    #+END_SRC

Welp, one more bug to fix.

@shosti: sorry about leading you in the wrong direction.  Would you
revert the PR to Emacs-comparatible behavior?  Maybe later, if @jgm
agrees, we could add a reader-option to allow nesting, your current
changes would come in handy then.